pkg/actors/actors.go

```diff
@@ -765,243 +748,43 @@ func (a *actorsRuntime) drainRebalancedActors() {
-				<-ttlTimer.C()
-			}
-		}()
-
-	L:
-		for {
-			select {
-			case <-nextTimer.C():
-				// noop
-			case <-ttlTimerC:
-				// proceed with reminder deletion
-				log.Infof("Reminder %s with parameters: dueTime: %s, period: %s has expired", reminderKey, reminder.DueTime, reminder.Period)
-				ttlTimer = nil
-				break L
-			case <-stopChannel:
-				// reminder has been already deleted
-				log.Infof("Reminder %s with parameters: dueTime: %s, period: %s has been deleted", reminderKey, reminder.DueTime, reminder.Period)
-				return
-			}
-
-			_, exists := a.activeReminders.Load(reminderKey)
-			if !exists {
-				log.Error("Could not find active reminder with key: " + reminderKey)
-				nextTimer = nil
-				return
-			}
-
-			// If all repetitions are completed, delete the reminder and do not execute it
-			if reminder.RepeatsLeft() == 0 {
-				log.Info("Reminder " + reminderKey + " has been completed")
-				nextTimer = nil
-				break L
-			}
-
-			err = a.executeReminder(reminder, false)
-			diag.DefaultMonitoring.ActorReminderFired(reminder.ActorType, err == nil)
-			if err != nil {
-				if errors.Is(err, ErrReminderCanceled) {
-					// The handler is explicitly canceling the timer
-					log.Debug("Reminder " + reminderKey + " was canceled by the actor")
-					nextTimer = nil
-					break L
-				} else {
-					log.Errorf("Error while executing reminder %s: %v", reminderKey, err)
-				}
-			}
-
-			_, exists = a.activeReminders.Load(reminderKey)
-			if exists {
-				err = a.updateReminderTrack(context.TODO(), reminderKey, reminder.RepeatsLeft(), reminder.NextTick(), eTag)
-				if err != nil {
-					log.Errorf("Error updating reminder track for reminder %s: %v", reminderKey, err)
-				}
-				track, gErr := a.getReminderTrack(context.TODO(), reminderKey)
-				if gErr != nil {
-					log.Errorf("Error retrieving reminder %s: %v", reminderKey, gErr)
-				} else {
-					eTag = track.Etag
-				}
-			} else {
-				log.Error("Could not find active reminder with key: " + reminderKey)
-				nextTimer = nil
-				return
-			}
-
-			if reminder.TickExecuted() {
-				nextTimer = nil
-				break L
-			}
-
-			nextTimer.Reset(reminder.NextTick().Sub(a.clock.Now()))
+		if errors.Is(err, ErrReminderCanceled) {
+			// The handler is explicitly canceling the timer
+			log.Debug("Reminder " + reminder.ActorKey() + " was canceled by the actor")
+			return false
 		}
+		log.Errorf("Error invoking reminder on actor %s: %s", reminder.ActorKey(), err)
+	}
 
-		err = a.DeleteReminder(context.TODO(), &DeleteReminderRequest{
-			Name:      reminder.Name,
-			ActorID:   reminder.ActorID,
-			ActorType: reminder.ActorType,
-		})
-		if err != nil {
-			log.Errorf("error deleting reminder: %s", err)
-		}
-	}()
-
-	return nil
+	return true
 }
 
 // Executes a reminder or timer
-func (a *actorsRuntime) executeReminder(reminder *reminders.Reminder, isTimer bool) (err error) {
+func (a *actorsRuntime) executeReminder(reminder *internal.Reminder, isTimer bool) (err error) {
```

Comment:
Nit on naming: perhaps this can be called `doExecuteReminderOrTimer` (or something like that), so we can have `executeTimer` and `executeReminder` above (instead of `executeReminderWrapper`). Or maybe both "wrapper" functions could be called like `executeTimerFn` and `executeReminderFn`? Anything as long as it's consistent.

Reply:
Corrected.